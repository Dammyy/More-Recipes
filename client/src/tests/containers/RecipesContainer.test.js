import React from 'react';
import { shallow } from 'enzyme';
import { RecipesContainer, mapStateToProps, mapDispatchToProps } from
  '../../containers/RecipesContainer';

describe('RecipesContainer', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        recipesActions: { getRecipes: jest.fn(), },
        recipes: [],
        userId: 1
      };
      const component = shallow(<RecipesContainer {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('container functions', () => {
    describe('mapStateToProps', () => {
      it('returns the expected properties', () => {
        const state = {
          getIn: () => ({ toJS: jest.fn() })
        };
        const mstp = mapStateToProps(state);
        expect(mstp).toHaveProperty('userId');
        expect(mstp).toHaveProperty('recipes');
      });
    });
  });
  describe('Instance methods', () => {
    describe('Recipes Container instance methods', () => {
      test('', () => {
        const props = {
          recipesActions: {
            getRecipes: jest.fn(),
            deleteRecipe: jest.fn(),
          },
          recipes: [],
          handlePageClick: jest.fn(),
          selected: 1
        };
        const component = shallow(<RecipesContainer {...props} />);
        component.instance().getRecipes();
        component.instance().deleteRecipe();
        component.instance().handlePageClick(props.selected);
        expect(props.recipesActions.getRecipes).toHaveBeenCalled();
      });
    });
  });
  describe('mapDispatchToProps', () => {
    it('', () => {
      const dispatch = jest.fn();
      const mdtp = mapDispatchToProps(dispatch);
      expect(mdtp).toHaveProperty('recipesActions');
    });
  });
});
