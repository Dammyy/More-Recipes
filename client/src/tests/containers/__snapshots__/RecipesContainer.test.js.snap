// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesContainer Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipesContainer
    recipes={Array []}
    recipesActions={
      Object {
        "getRecipes": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
      }
    }
    userId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RecipesList
          deleteRecipe={[Function]}
          recipes={undefined}
          userId={1}
        />,
        <PaginationBoxView
          activeClassName="active"
          breakClassName="break-me"
          breakLabel="..."
          containerClassName="pagination"
          disableInitialCallback={false}
          disabledClassName="disabled"
          marginPagesDisplayed={2}
          nextClassName="next"
          nextLabel="next"
          onPageChange={[Function]}
          pageCount={10}
          pageRangeDisplayed={5}
          previousClassName="previous"
          previousLabel="previous"
          subContainerClassName="pages pagination"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deleteRecipe": [Function],
          "recipes": undefined,
          "userId": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeClassName": "active",
          "breakClassName": "break-me",
          "breakLabel": "...",
          "containerClassName": "pagination",
          "disableInitialCallback": false,
          "disabledClassName": "disabled",
          "marginPagesDisplayed": 2,
          "nextClassName": "next",
          "nextLabel": "next",
          "onPageChange": [Function],
          "pageCount": 10,
          "pageRangeDisplayed": 5,
          "previousClassName": "previous",
          "previousLabel": "previous",
          "subContainerClassName": "pages pagination",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RecipesList
            deleteRecipe={[Function]}
            recipes={undefined}
            userId={1}
          />,
          <PaginationBoxView
            activeClassName="active"
            breakClassName="break-me"
            breakLabel="..."
            containerClassName="pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={2}
            nextClassName="next"
            nextLabel="next"
            onPageChange={[Function]}
            pageCount={10}
            pageRangeDisplayed={5}
            previousClassName="previous"
            previousLabel="previous"
            subContainerClassName="pages pagination"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "deleteRecipe": [Function],
            "recipes": undefined,
            "userId": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "active",
            "breakClassName": "break-me",
            "breakLabel": "...",
            "containerClassName": "pagination",
            "disableInitialCallback": false,
            "disabledClassName": "disabled",
            "marginPagesDisplayed": 2,
            "nextClassName": "next",
            "nextLabel": "next",
            "onPageChange": [Function],
            "pageCount": 10,
            "pageRangeDisplayed": 5,
            "previousClassName": "previous",
            "previousLabel": "previous",
            "subContainerClassName": "pages pagination",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
